### PR TITLE
[CALCITE-2006] Push IS NULL and IS NOT NULL predicates to Druid

### PR DESCRIPTION
**Pull request by jcamachor**

### CODE REVIEW

File: druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java [hunk 1]
```diff
@@ -1151,10 +1153,15 @@ private JsonFilter translateFilter(RexNode e) {
       case LESS_THAN_OR_EQUAL:
       case IN:
       case BETWEEN:
+      case IS_NULL:
+      case IS_NOT_NULL:
         call = (RexCall) e;
         int posRef;
         int posConstant;
-        if (RexUtil.isConstant(call.getOperands().get(1))) {
+        if (call.getOperands().size() == 1) { // IS NULL and IS NOT NULL
```
Comment (b-slim) [inline druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java]: is this the only case where the filter is `is null` or `is not null`?

File: druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java [hunk 2]
```diff
@@ -3242,6 +3244,36 @@ private void testCountWithApproxDistinct(boolean approx, String sql, String expe
...
+        + "where \"product_id\" is null";
+    final String druidQuery =
+        "{'queryType':'timeseries','dataSource':'foodmart','descending':false,'granularity':'all',"
+            + "'filter':{'type':'selector','dimension':'product_id','value':null},"
+            + "'aggregations':[{'type':'doubleMin','name':'A','fieldName':'store_cost'}],"
+            + "'intervals':['1900-01-09T00:00:00.000/2992-01-10T00:00:00.000'],"
+            + "'context':{'skipEmptyBuckets':true}}";
+    sql(sql, FOODMART)
+        .queryContains(druidChecker(druidQuery))
+        .returnsCount(0);
```
Comment (b-slim) [inline druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java]: do we have a column in this dataset where values are null? if not it is fine I guess.

File: druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java (hunk 1 above)
Comment (jcamachor) [reply druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java]: This block refers to following operators: EQUALS, NOT_EQUALS, GREATER_THAN, GREATER_THAN_OR_EQUAL, LESS_THAN, LESS_THAN_OR_EQUAL, IN, BETWEEN, IS_NULL, IS_NOT_NULL. Of those, only IS_NULL and IS_NOT_NULL can have 1 operand.

File: druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java [hunk 3]
```diff
@@ -3242,6 +3244,36 @@ private void testCountWithApproxDistinct(boolean approx, String sql, String expe
...
+  }
+
+  @Test public void testIsNotNull() {
```
Comment (b-slim) [inline druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java]: I guess this answers to https://github.com/apache/calcite/pull/546/files#r143879698

File: druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java (hunk 1 above)
Comment (b-slim) [reply druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java]: Ok good to know

**Comment by b-slim** [comment]
👍

File: druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java (hunk 2 above)
Comment (jcamachor) [reply druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java]: We should be adding new tests once Druid version in the VM is updated to latest (with full NULL support).

**Pull request closed without merging**